### PR TITLE
Bug fix/cloud config overwrite

### LINKED ISSUE #9: Some settings aren't making it to the OpenVPN config.
@felddy commented on [Fri Mar 06 2020](https://github.com/cisagov/ansible-role-openvpn/issues/4)

```
push "route 10.128.0.0 255.128.0.0"
server 10.128.248.0 255.255.248.0
push "dhcp-option DOMAIN cool.cyber.dhs.gov"
push "dhcp-option DNS 10.128.0.2"
```

### PR DESCRIPTION
**Pull request by felddy**
# 

## 🗣 Description

This fixes the cloud-config not properly rendering during boot.  It required a merge command for each of the cloud-config parts.

Fixes #9 

## 💭 Motivation and Context

This was a bug.  The OpenVPN `primary.conf` file was not being rendered, which means the OpenVPN server was not properly configured. (It would use the defaults from the ansible-role)

## 🧪 Testing

Deployed new instance successfully.  Configurations were applied as written.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
This looks correct- did you re-deploy the OpenVPN instance with this new Terraform to confirm that the bug was indeed fixed?
❓ 🤕

**Comment by felddy** [comment]
Not yet.  There is no way this can fail.

**Review by mcdonnnj — APPROVED** [review]
> Testing
> Not much.

:shipit:

File: cloud_init.tf [hunk 1]
```diff
@@ -4,17 +4,24 @@ data "template_cloudinit_config" "cloud_init_tasks" {
...
   part {
-    filename     = "openvpn-config"
+    filename     = "openvpn-config.yml"
     content_type = "text/cloud-config"
     content = templatefile(
       "${path.module}/cloudinit/openvpn-config.tpl.yml", {
         private_networks = var.private_networks
         client_network   = var.client_network
     })
+    merge_type = "list(append)+dict(recurse_array)+str()"
```
Comment (jsf9k) [inline cloud_init.tf:24]: Are the two `merge_type` lines necessary?  I don't remember the innards of cloud-init wizardry right now.

If they are required, they probably warrant a comment.

**Review by jsf9k — CHANGES_REQUESTED** [review]
Minor request for a comment; otherwise this looks good.

File: cloud_init.tf (hunk 1 above)
Comment (dav3r) [reply cloud_init.tf:24]: I also seem to recall us only needing one `merge_type` line in the past for a similar situation.

Comment (jsf9k) [reply cloud_init.tf:24]: @felddy, after you reminded me I am onboard with the `merge_type`.  I'd like to see comments there, though, so I don't need a personal refresher class in cloud-init next time.

Comment (felddy) [reply cloud_init.tf:24]: See: 
https://github.com/search?q=org%3Acisagov+merge_type&type=Code

**Review by jsf9k — APPROVED** [review]

File: cloud_init.tf (hunk 1 above)
Comment (felddy) [reply cloud_init.tf:24]: I added 3e613e9 for you @jsf9k .  Good idea.

**Pull request merged**